### PR TITLE
v156: Admin products and categories facelift/refactoring

### PR DESCRIPTION
**Pull request by Zen4All**
see #1583 for more information

@drbyte edit:
Closes #1631

### CODE REVIEW

**Comment by drbyte** [comment]
@Zen4All ... thoughts on readiness for review/merge?

**Comment by Zen4All** [comment]
Hoping to be ready end of this week

**Comment by drbyte** [comment]
I hope to review this tomorrow.

**Comment by Zen4All** [comment]
Okay, I have some latest css changes am bug-fixes, will commit them this evening. After that, there may be one or two defines that are missing, but that is it.

**Comment by drbyte** [comment]
Have you any thoughts on how best to handle the floating Preview button?

image 2018-03-30 at 3 37 25 pm
image 2018-03-30 at 3 38 38 pm

**Comment by Zen4All** [comment]
Sorry, I do not understand what you want me to do.

**Comment by drbyte** [comment]
I was wondering your opinion on how much the overlap may be a problem for users, and what options could be presented as a ‘solution’. 

In practical use I’m guessing it’s only a minor annoyance. While there’s always someone who will complain, it can always be addressed later.

**Comment by Zen4All** [comment]
The idea of making a floating button came from one of my customers who has a 3 language store. He got tired of scrolling up or down. Since then I implemented this on multiple stores, and nobody complained about the overlap.
One option could be to make the space between the bottom of the screen an bottom of the button.

File: zc_install/sql/updates/mysql_upgrade_zencart_156.sql [hunk 1]
```diff
@@ -127,7 +127,16 @@ SELECT products_attributes_id FROM products_attributes WHERE products_id NOT IN
...
+## alter admin_pages for new product listing pages
+UPDATE admin_pages
+SET language_key = 'BOX_CATALOG_CATEGORY',
+    main_page = 'FILENAME_CATEGORY_PRODUCT_LISTING',
+    display_on_menu = 'N',
+    sort_order = 18
+WHERE page_key = 'categories';
 
+INSERT INTO admin_pages (page_key, language_key, main_page, page_params, menu_key, display_on_menu, sort_order)
+VALUES ('categoriesProductListing', 'BOX_CATALOG_CATEGORIES_PRODUCTS', 'FILENAME_CATEGORY_PRODUCT_LISTING', '', 'catalog', 'Y', 1);
```
Comment (drbyte) [inline zc_install/sql/updates/mysql_upgrade_zencart_156.sql]: Why the insertion of a new key, instead of renaming-and-hiding?

Comment (Zen4All) [reply zc_install/sql/updates/mysql_upgrade_zencart_156.sql]: I always thought the pages had to be in the admin_pages  table, so you can allow or disallow them in the profiles.
That is why I added a new item for the listing, and updated the old item so it represents the category page.

**Comment by zcwilt** [comment]
:+1:

**Pull request merged**